include/cetl/pf17/string_view.hpp

```diff
@@ -0,0 +1,295 @@
+/// @file
+/// Defines the C++17 std::variant type and several related entities.
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_PF17_STRING_VIEW_HPP_INCLUDED
+#define CETL_PF17_STRING_VIEW_HPP_INCLUDED
+
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+#include <exception>  // We need this even if exceptions are disabled for std::terminate.
+#include <stdexcept>
+#include <string>
+
+namespace cetl
+{
+namespace pf17
+{
+
+class string_view
```

Comment:
In the standard, string_view is just an alias for basic_string_view<char> so we need to start with this template.

Reply:
ok, I can do it. My thinking was that it will add more work while we most probably will use `char` specialization only.